carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/Screenshot.java

```diff
@@ -387,6 +405,37 @@ private static void uploadToAmazonS3(String test, String fullScreenPath, String
         AmazonS3Manager.getInstance().put(screenshotBucket, key, fullScreenPath, metadata);
     }
 
+    /**
+     * Upload screenshot file to Amazon S3 using zafira client
+     * @param screenshot - existing screenshot {@link File}
+     */
+    private static void uploadToAmazonS3(File screenshot) {
+        if (!Configuration.getBoolean(Parameter.S3_SAVE_SCREENSHOTS_V2)) {
+            LOGGER.debug("there is no sense to continue as saving screenshots onto S3 is disabled.");
+            return;
+        }
+        String correlationId = UUID.randomUUID().toString();
+        TestType test = ZafiraListener.getTestbythread().get(Thread.currentThread().getId());
+        try {
+            ZafiraMessager.<MetaInfoMessage>custom(MetaInfoLevel.META_INFO, new MetaInfoMessage()
+                    .addHeader("AMAZON_PATH", null)
+                    .addHeader("AMAZON_PATH_CORRELATION_ID", correlationId));
+            executorService.execute(() -> {
+                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://demo.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS1141)